client/src/utils/utils.js

```diff
@@ -0,0 +1,12 @@
+//author medium.com/@kartikag01
```

Comment:
If we are abstracting a utility function, we should properly name the file (like `debounceFunction.js`) and give it a simple comment outlining its usage and purpose.

Reply:
I added comments to debounce function.  We may end up having more than one general utility function so I think I'll keep it in util file for now.

Reply:
What does this function do?

Reply:
This function takes a function as the first argument and time in milliseconds as second argument and returns a function that runs the original function (argument 1) a maximum of once in a period of time supplied as (argument 2).  Here is an article explaining debouncing in detail: https://levelup.gitconnected.com/debounce-in-javascript-improve-your-applications-performance-5b01855e086